### PR TITLE
#7651 - VRF Condenser not simulated when one VRF TU not in Equipment List

### LINKED ISSUE #7651: VRF Condenser not simulated when one VRF TU not in Equipment List
Issue overview
--------------
AirConditionar:VariableRefrigerantFlow with multiple terminal units is not simulated when a ZoneHVAC:TerminalUnit:VariableRefrigerantFlow terminal unit is not listed in the corresponding ZoneHVAC:EquipmentList.

In code, this unit IsSimulated property remains as false and the the condenser is not simulated according to code:
https://github.com/NREL/EnergyPlus/blob/775eef1f74efd67e959d4b15eb5061a3e3b80198/src/EnergyPlus/HVACVariableRefrigerantFlow.cc#L370-L386

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------
 - Fixes #7651 - VRF Condenser not simulated when one VRF TU not in Equipment List
 - As stated on the issue itself (cf: https://github.com/NREL/EnergyPlus/issues/7651#issuecomment-579754489) I have started by creating a test file on EnergyPlusDevSupport [here](https://github.com/NREL/EnergyPlusDevSupport/tree/master/DefectFiles/7000s/7651) and realized it's pretty much already addressed
 - I added a specific unit test to close #7651 though
 - and avoid a redundant check: this portion already does what CheckZoneEquipmentList does (and more), and stores the boolean result on `VRFTU(TUIndex).isInZone`, so no need to call CheckZoneEquipmentList later.
 
https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/HVACVariableRefrigerantFlow.cc#L5206-L5220

For ref `CheckZoneEquipmentList` is implemented here:

https://github.com/NREL/EnergyPlus/blob/0dfe78ae4e754cf7e72fcc465e31c4b2aa771bdf/src/EnergyPlus/DataZoneEquipment.cc#L1192-L1236

Note: Some trailing spaces that were in the file are producing whitespace diff, so append `?w=1` to the compare: https://github.com/NREL/EnergyPlus/pull/7995/files?w=1

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/HVACVariableRefrigerantFlow.cc [hunk 1]
```diff
@@ -5542,8 +5542,7 @@ namespace HVACVariableRefrigerantFlow {
                         }
                     }
 
-                    if (VRFTU(TUIndex).isInAirLoop || VRFTU(TUIndex).isInOASys ||
-                        CheckZoneEquipmentList(DataHVACGlobals::cVRFTUTypes(VRFTU(TUIndex).VRFTUType_Num), VRFTU(TUIndex).Name))
+                    if (VRFTU(TUIndex).isInAirLoop || VRFTU(TUIndex).isInOASys || VRFTU(TUIndex).isInZone)
```
Comment (jmarrec) [inline src/EnergyPlus/HVACVariableRefrigerantFlow.cc:5657]: Avoid a redundant check

File: tst/EnergyPlus/unit/HVACVariableRefrigerantFlow.unit.cc [hunk 2]
```diff
@@ -14239,5 +14239,722 @@ TEST_F(EnergyPlusFixture, VRF_MinPLR_and_EIRfPLRCruveMinPLRInputsTest)
...
+        "    20.0000;                 !- Zone Terminal Unit Off Parasitic Electric Energy Use {W}",
+
+        // This TU is NOT on any ZoneHVACEquipmentList
+        "  ZoneHVAC:EquipmentList,",
+        "    Level1:Office1 Equipment,!- Name",
+        "    SequentialLoad;          !- Load Distribution Scheme",
+        // "    ZoneHVAC:TerminalUnit:VariableRefrigerantFlow,  !- Zone Equipment 1 Object Type",
+        // "    Level1:Office1 VRF Indoor Unit,  !- Zone Equipment 1 Name",
+        // "    1,                       !- Zone Equipment 1 Cooling Sequence",
+        // "    1;                       !- Zone Equipment 1 Heating or No-Load Sequence",
```
Comment (jmarrec) [inline tst/EnergyPlus/unit/HVACVariableRefrigerantFlow.unit.cc:14307]: Unit Test: the TU is purposefully not placed on a `ZoneHVAC:EquipmentList`

File: tst/EnergyPlus/unit/HVACVariableRefrigerantFlow.unit.cc [hunk 3]
```diff
@@ -14239,5 +14239,722 @@ TEST_F(EnergyPlusFixture, VRF_MinPLR_and_EIRfPLRCruveMinPLRInputsTest)
...
+
+
+    InitVRF(state, VRFTUNum, ZoneNum, FirstHVACIteration, OnOffAirFlowRatio, QZnReq); // Initialize all VRFTU related parameters
+
+    std::string const error_string = delimited_string({
+        "   ** Severe  ** InitVRF: VRF Terminal Unit = [ZoneHVAC:TerminalUnit:VariableRefrigerantFlow,LEVEL1:OFFICE1 VRF INDOOR UNIT] is not on any ZoneHVAC:EquipmentList, AirloopHVAC or AirLoopHVAC:OutdoorAirSystem:EquipmentList.  It will not be simulated.",
+        "   **   ~~~   ** ...The VRF AC System associated with this terminal unit may also not be simulated."
+    });
+
+    EXPECT_TRUE(compare_err_stream(error_string, true));
```
Comment (jmarrec) [inline tst/EnergyPlus/unit/HVACVariableRefrigerantFlow.unit.cc:14954]: Unit Test: check that you are indeed getting the `Severe` you expected to see.

**Comment by mitchute** [comment]
CI is happy and the new unit test captures the intended behavior nicely. Merging.

**Pull request merged**